eth-contracts/test/upgradeProxy.test.js

```diff
@@ -41,90 +37,117 @@ contract('Upgrade proxy test', async (accounts) => {
+  it('upgrade proxy to StakingUpgraded + call newFunction()', async () => {
+    proxy = await AdminUpgradeabilityProxy.new(
+      staking0.address,
+      proxyAdminAddress,
+      stakingInitializeData,
+      { from: proxyDeployerAddress }
+    )
+
+    // assert proxy.newFunction() not callable before upgrade
+    staking = await StakingUpgraded.at(proxy.address)
+    await _lib.assertRevert(staking.newFunction.call({ from: proxyDeployerAddress }))
+
+    await proxy.upgradeTo(stakingUpgraded.address, { from: proxyAdminAddress })
+    assert.equal(await proxy.implementation.call({ from: proxyAdminAddress }), stakingUpgraded.address)
+
+    // assert proxy.newFunction() call succeeds after upgrade
+    staking = await StakingUpgraded.at(proxy.address)
+    const newFunctionResp = await staking.newFunction.call({ from: proxyDeployerAddress })
+    assert.equal(newFunctionResp, 5)
   })
 
-  describe('initial staking function provided', function () {
+  describe('Confirm staking logic', function () {
     beforeEach(async function () {
-      const spAccount1 = accounts[1]
-      const spAccount2 = accounts[2]
+      const spAccount1 = accounts[3]
+      const spAccount2 = accounts[4]
 
       // Transfer 1000 tokens to accounts[1] and accounts[2]
       await token.transfer(spAccount1, 1000, { from: treasuryAddress })
       await token.transfer(spAccount2, 1000, { from: treasuryAddress })
 
-      await proxy.upgradeToAndCall(impl0.address, initializeData, { from: proxyOwner })
+      proxy = await AdminUpgradeabilityProxy.new(
+        staking0.address,
+        proxyAdminAddress,
+        stakingInitializeData,
+        { from: proxyDeployerAddress }
+      )
 
       // Permission test address as caller
-      await staking0.setStakingOwnerAddress(testStakingCallerAddress, { from: treasuryAddress })
+      staking = await Staking.at(proxy.address)
+      await staking.setStakingOwnerAddress(stakingOwnerAddress, { from: treasuryAddress })
     })
 
     it('upgrade and confirm initial staking state at proxy', async () => {
-      const implementation = await proxy.implementation()
-      assert.equal(implementation, impl0.address)
-      assert.equal(await staking0.hasInitialized(), true, 'contract has not been initialized')
-      assert.equal(await staking0.token(), token.address, 'Token is wrong')
-      assert.equal((await staking0.totalStaked()).valueOf(), 0, 'Initial total staked amount should be zero')
-      assert.equal(await staking0.supportsHistory(), true, 'history support should match')
-    })
-
-    it('fails to invoke unknown function', async () => {
-      const implementation = await proxy.implementation()
-      assert.equal(implementation, impl0.address)
-      _lib.assertRevert(staking1.testFunction())
+      assert.equal(await proxy.implementation.call({ from: proxyAdminAddress }), staking0.address)
+      assert.equal(await staking.initializerRan.call({ from: accounts[3]}), true, 'contract has not been initialized')
```

Comment:
yeah not sure we need initializerRan at all per prior comment

Reply:
fixed